src/libraries/Common/src/Interop/Linux/cgroups/Interop.cgroups.cs

```diff
@@ -128,16 +148,39 @@ internal static bool TryReadMemoryValueFromFile(string path, out ulong result)
         /// <summary>Find the cgroup path for the specified subsystem.</summary>
         /// <param name="subsystem">The subsystem, e.g. "memory".</param>
         /// <returns>The cgroup path if found; otherwise, null.</returns>
-        private static string? FindCGroupPath(string subsystem, out CGroupVersion version)
+        private static string? FindCGroupPath(string subsystem)
         {
-            if (TryFindHierarchyMount(subsystem, out version, out string? hierarchyRoot, out string? hierarchyMount) &&
+            if (s_cgroupVersion == CGroupVersion.None)
+            {
+                return null;
+            }
+
+            if (TryFindHierarchyMount(subsystem, out string? hierarchyRoot, out string? hierarchyMount) &&
                 TryFindCGroupPathForSubsystem(subsystem, out string? cgroupPathRelativeToMount))
             {
                 // For a host cgroup, we need to append the relative path.
-                // In a docker container, the root and relative path are the same and we don't need to append.
-                return (hierarchyRoot != cgroupPathRelativeToMount) ?
-                    hierarchyMount + cgroupPathRelativeToMount :
-                    hierarchyMount;
+                // The root and cgroup path can share a common prefix of the path that should not be appended.
+                // Example 1 (docker):
+                // hierarchyMount:               /sys/fs/cgroup/cpu
+                // hierarchyRoot:                /docker/87ee2de57e51bc75175a4d2e81b71d162811b179d549d6601ed70b58cad83578
+                // cgroupPathRelativeToMount:    /docker/87ee2de57e51bc75175a4d2e81b71d162811b179d549d6601ed70b58cad83578/my_named_cgroup
+                // append to the cgroupPath:     /my_named_cgroup
+                // final cgroupPath:             /sys/fs/cgroup/cpu/my_named_cgroup
+                //
+                // Example 2 (out of docker)
+                // hierarchyMount:               /sys/fs/cgroup/cpu
+                // hierarchyRoot:                /
+                // cgroupPathRelativeToMount:    /my_named_cgroup
+                // append to the cgroupPath:     /my_named_cgroup
+                // final cgroupPath:             /sys/fs/cgroup/cpu/my_named_cgroup
+
+                int commonPathPrefixLength = hierarchyRoot.Length;
+                if ((commonPathPrefixLength == 1) || (hierarchyRoot != cgroupPathRelativeToMount))
+                {
+                    commonPathPrefixLength = 0;
+                }
+
+                return hierarchyMount + cgroupPathRelativeToMount.Substring(commonPathPrefixLength);
```

Comment:
This can be:
```suggestion
                return string.Concat(hierarchyMount, cgroupPathRelativeToMount.AsSpan(commonPathPrefixLength));
```
to avoid the interim substring allocation.

Reply:
Thanks!